### PR TITLE
feat: Add comprehensive Conditional Block metadata test cases

### PR DESCRIPTION
**Pull request by bastelbude1**
- Added 6 test cases for type=conditional blocks
- Tests cover true/false branching, nested conditionals, exit codes, variables, and edge cases
- All tests validated and passing with intelligent_test_runner.py

Test Coverage:
1. Basic conditional (true/false branches based on stdout)
2. False branch execution (validates correct branch selection)
3. Nested conditionals (conditionals within conditional branches)
4. Exit code conditions (using exit_code and success conditions)
5. Variable conditions (global variables in conditions)
6. Edge cases (empty branches, missing tasks)

Note: Conditional branch tasks (e.g., tasks 10, 11 in if_true_tasks) do not appear in execution_path as they are logged as sub-tasks (Task 1-10). This is expected behavior for conditional blocks.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: test_cases/functional/metadata_conditional_empty_branch_test.txt [hunk 1]
```diff
@@ -0,0 +1,36 @@
+# TEST_METADATA: {"description": "Conditional Block test - empty branches and edge cases", "test_type": "positive", "expected_exit_code": 0, "expected_success": true, "expected_execution_path": [0, 3], "expected_skipped_tasks": []}
```
[the remark this answers is not in the document]
Comment (bastelbude1) [reply test_cases/functional/metadata_conditional_empty_branch_test.txt:1]: to the tasker logic and conditional block, from my point of view it makes absolutely no sense to allow empty on success or on failure: # Conditional with empty true branch (edge case)
task=1
type=conditional
condition=@0_stdout@=NOMATCH
if_true_tasks=
if_false_tasks= . if an conditional block is used, the definition of at least one task per if_true_tasks and if_false_tasks is mandatory, so first of all, task validation should report this as an error
 and if we would  use ignore task validation, we should stop with an fatal error, as the correct workflow can not be followed. think hard how to implement these fixes correctly, without breaking any
code

afterwards, we need to update the test case to reflect the new expected_execution_path
